backend/packages/Upgrade/src/api/services/AnalyticsService.ts

```diff
@@ -199,24 +206,26 @@ export class AnalyticsService {
             enrollmentCompleteDate: item.enrollmentCompleteDate,
             details: [],
           };
+          experimentMap.set(item.experimentId, experiment);
           acc.push(experiment);
         }
-        experiment.details.push({
+        // add the design details for the experiment for each experimentQueryResult item
+        experimentMap.get(item.experimentId)?.details.push({
```

Comment:
If this was overwriting the details for existing experiments, can't we fix that by simply closing the `if` statement on line 221 (old 212), rather than creating the map structure?

Reply:
ye we had that in the old code, I felt using a Map would be more efficient. You want me to revert this part? Wont make much of a difference though.